### PR TITLE
chore(deps): update yansi to v1.0

### LINKED ISSUE #7709: Update to yansi 1.0
### Component

Other (please describe)

### Describe the feature you would like

[`yansi 1.0`](https://crates.io/crates/yansi) was recently released with a much better API, by turning the `Paint` struct into a trait implemented on all types. This will also require updating it in [`foundry-compilers`](https://github.com/foundry-rs/compilers).

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dcb9**
closes #7709

### CODE REVIEW

File: crates/common/Cargo.toml [hunk 1]
```diff
@@ -60,7 +60,7 @@ tokio = "1"
 tracing.workspace = true
 url = "2"
 walkdir = "2"
-yansi = "0.5"
+yansi = "1.0"
```
Comment (mattsse) [inline crates/common/Cargo.toml:63]: can we move this to workspace deps (the root Cargo.toml) and then use yansi.workspace here?

**Review by mattsse — CHANGES_REQUESTED** [review]
awesome, I think everything looks correct.
ptal @DaniPopes 

1 dep nit

File: crates/common/Cargo.toml (hunk 1 above)
Comment (dcb9) [reply crates/common/Cargo.toml:63]: yes I think it would be better cause all crates should share same foundation to keep them as a unity

**Review by DaniPopes — APPROVED** [review]
Thanks!

**Review by mattsse — APPROVED** [review]

**Pull request merged**